engine/src/dskmac.cpp

```diff
@@ -4627,9 +4627,63 @@ struct MCMacDesktop: public MCSystemInterface, public MCMacSystemService
         if (t_cf_document != NULL)
             CFRelease(t_cf_document);
     }
-    
+
+#define ENVIRONMENT_TRANSLATED_SCRIPT \
+	"return line 1 of shell(\"sysctl -in sysctl.proc_translated\")"
+
+#define APPLESCRIPT_SCRIPT \
+	"local tTempFolder;" \
+	"put the tempname into tTempFolder;" \
+	"create folder tTempFolder;" \
+	"local tStdout, tStderr, tScript;" \
+	"put tTempFolder & \"/stdout.txt\" into tStdout;" \
+	"put tTempFolder & \"/stderr.txt\" into tStderr;" \
+	"put tTempFolder & \"/script.scpt\" into tScript;" \
+	"put textEncode(param(1), \"utf8\") into url (\"binfile:\" & tScript);" \
+	"get shell(format(\"arch -arm64 osascript %s 2>%s 1>%s\", tScript, tStderr, tStdout));" \
+	"local tResult;" \
+	"put the result into tResult;" \
+	"delete file tScript;" \
+	"local tValue;" \
+	"if tResult is 0 then;" \
+		"put url (\"binfile:\" & tStdout) into tValue;" \
+		"if the last char of tValue is return then;" \
+			"delete the last char of tValue;" \
+		"end if;" \
+	"else;" \
+		"put \"execution error\" into tValue;" \
+	"end if;" \
+	"delete file tStdout;" \
+	"delete file tStderr;" \
+	"switch tValue;" \
+	"case \"execution error\";" \
+	"case empty;" \
+		"return tValue;" \
+	"default;" \
+		"return \"{\" & tValue & \"}\";" \
+	"end switch"
+
     virtual void DoAlternateLanguage(MCStringRef p_script, MCStringRef p_language)
     {
+		uinteger_t t_environment_translated = 0;
+		if (MCStringIsEqualToCString(p_language, "AppleScript", kMCStringOptionCompareCaseless))
+		{
+			MCresult->clear();
```

Comment:
We don't need to use shell to determine whether we are running under a translated environment we can easily do that in C - https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment:
```
int processIsTranslated() {
   int ret = 0;
   size_t size = sizeof(ret);
   if (sysctlbyname("sysctl.proc_translated", &ret, &size, NULL, 0) == -1) 
   {
      if (errno == ENOENT)
         return 0;
      return -1;
   }
   return ret;
}
```
Also we can cache the result of testing for rosetta in a local static as it isn't going to change for the lifetime of the process :)